### ISSUE TITLE
groupBitmapOr result incorrect

### ISSUE THREAD

**Issue by matthew-lei** [ClickHouse/ClickHouse#31335]
**Describe the unexpected behaviour**
groupBitmapOr result 0 but bitmapCardinality record result > 0 in same condition 

**How to reproduce**
```sql
CREATE TABLE ucube.ucube_source_level_tag_uv2 on cluster ucube_ck_cluster
(
    `partition_time` Int64,
    `source` Int64,
    `level` Int64,
    `tag_id` Int64,
    `dimension` Int64,
    `name` String,
    `full_id` String,
    `user_set` Array(UInt32),
    `user_uv` AggregateFunction(groupBitmap, UInt32) MATERIALIZED bitmapBuild(user_set) COMMENT 'userid bitmap'
)
ENGINE = MergeTree
PARTITION BY partition_time
PRIMARY KEY (source, level, tag_id, dimension, full_id)
ORDER BY (source, level, tag_id, dimension, full_id)
SETTINGS index_granularity = 8192;
-- create distribute table
CREATE TABLE ucube.ucube_source_level_tag_uv2_view on cluster ucube_ck_cluster AS ucube.ucube_source_level_tag_uv2 ENGINE=Distributed(ucube_ck_cluster, ucube, ucube_source_level_tag_uv2, rand());
```
then insert into tables 20 000 000 records and each record user_set.length < 10000
1. calculate each array cardinality
```sql
select bitmapCardinality(user_uv) from ucube_source_level_tag_uv2_view where full_id = '7_61_146';
```
136 rows in set and each result > 0
![image](https://user-images.githubusercontent.com/68599927/141444906-0590cf0f-add1-49da-b56f-8f10849c7bef.png)
2. But when i calculate all array cardinality by groupBitmapOr in same condition, it returns 0
```sql
select groupBitmapOr(user_uv) from ucube_source_level_tag_uv2_view where full_id = '7_61_146';
```
![image](https://user-images.githubusercontent.com/68599927/141445109-2b90f881-4cc4-4ad9-831e-07c16bf2ec67.png)
* Which ClickHouse server version to use
21.9.2.17

**Expected behavior**
groupBitmapOr result may approximate to Sum(bitmapCardinality), because element in each record `user_set` array is different in the condition

**Error message and/or stacktrace**
No Error

**Comment by matthew-lei** [comment]
new discovery:
direct cal groupBitmapOr(user_uv) return 0
```sql
select groupBitmapOr(user_uv) from ucube_source_level_tag_uv2_view where full_id = '7_61_146';
```
![image](https://user-images.githubusercontent.com/68599927/141716161-89239428-9b9a-4d65-b66f-b43fc9ee212f.png)

but cal groupBitmpaOr(user_uv) after select user_uv field, it returns real count
```sql
select groupBitmapOr(user_uv) from (select user_uv from ucube_source_level_tag_uv2_view where full_id = '7_61_146');
```
![image](https://user-images.githubusercontent.com/68599927/141716204-8200703d-5f04-4cd8-a716-c09b88b2d0f1.png)

**Comment by abel-cheng** [comment]
Solved in [#32529 ](https://github.com/ClickHouse/ClickHouse/pull/32529)
